### PR TITLE
Add documentation on release procedure

### PR DESCRIPTION
**Pull request by danobi**
Documenting the procedure should make things more consistent.

### CODE REVIEW

File: docs/release_process.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   ```
+   where `${PREVIOUS_VER}` is the last release tag, eg `v0.9.3`. Please see previous
+   releases for the final formatting.
+1. Update `bpftrace_VERSION_MAJOR`, `bpftrace_VERSION_MINOR`, and
+   `bpftrace_VERSION_PATCH` in `CMakeLists.txt` to the target version.
+1. Tag a release. We do this in the github UI by clicking "releases" (on same line
+   as "commits"), then "Draft a new release". The tag version and release title
+   should be the same and in `vX.Y.Z` format. The tag description should
+   be the same as what you added to `CHANGELOG.md`.
+1. (optional) Announce the new release to iovisor-dev@lists.iovisor.org and
```
Comment (fbs) [inline docs/release_process.md]: I think we should either announce releases or just skip it. Having some versions announced and other released silently will be confusing, people won't be able to rely on the announce email going out and will still have to track github.

I think we should leave it out for now, once the change logs have improved a bit we can start sending them out.

File: docs/release_process.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+isn't a huge project yet so complicated branching models and release strategies
+more get in the way than provide order. If master is really busy or really buggy,
+the RM can choose to cut a release branch (titled `X.Y.Z_release`) to try and
+stabilize the code without including work in progress into the release.
+
+## Merging pull requests
+
+Please squash + rebase all pull requests (with no merge commit). In other words,
+there should be one commit in master per pull request. This makes generating
+changelogs both trivial and precise with the least amount of noise.
```
Comment (fbs) [inline docs/release_process.md]: +1 for doing this on the smaller merge requests, but you loose quite some context on the bigger ones if you do this. 

When the author has a cleanly written history AND the change is sufficiently complex we should allow rebase merges (without squash). But I've not found a good way to define sufficiently yet...

**Comment by ajor** [comment]
Nice write up @danobi!

File: docs/release_process.md (hunk 1 above)
Comment (danobi) [reply docs/release_process.md]: > I think we should leave it out for now, once the change logs have improved a bit we can start sending them out.

Sounds good. We can discuss again when we get close to 1.0

**Comment by danobi** [comment]
Updated. Please let me know if there's any other comments.

**Pull request merged**